src/evo/creditpool.cpp

```diff
@@ -213,21 +212,38 @@ CCreditPoolManager::CCreditPoolManager(CEvoDB& _evoDb)
 
 CCreditPoolDiff::CCreditPoolDiff(CCreditPool starter, const CBlockIndex *pindex, const Consensus::Params& consensusParams) :
     pool(std::move(starter)),
-    pindex(pindex)
+    pindex(pindex),
+    params(consensusParams)
 {
     assert(pindex);
 }
 
+void CCreditPoolDiff::AddRewardRealloced(const CAmount reward) {
+    assert(MoneyRange(reward));
+    masternodeReward += reward;
+}
+
 bool CCreditPoolDiff::SetTarget(const CTransaction& tx, TxValidationState& state)
 {
     CCbTx cbTx;
     if (!GetTxPayload(tx, cbTx)) {
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-cbtx-payload");
     }
 
-    if (cbTx.nVersion == 3) {
-        targetBalance = cbTx.creditPoolBalance;
+    if (cbTx.nVersion != 3) return true;
+
+    targetBalance = cbTx.creditPoolBalance;
+
+
+    if (!llmq::utils::IsMNRewardReallocationActive(pindex)) return true;
+
+    CAmount blockReward = 0;
+    for (const CTxOut& txout : tx.vout) {
+        blockReward += txout.nValue;
     }
+    masternodeReward = 0.375 * GetMasternodePayment(cbTx.nHeight, blockReward, params.BRRHeight);
```

Comment:
can we get rid of this instance of the `0.375` as well?

Reply:
thanks for catch, completely missed this one. Please, check new commits